lib/jsdom/living/nodes/HTMLLabelElement-impl.js

```diff
@@ -4,9 +4,62 @@ const HTMLElementImpl = require("./HTMLElement-impl").implementation;
 const closest = require("../helpers/traversal").closest;
 
 class HTMLLabelElementImpl extends HTMLElementImpl {
+  constructor(args, privateData) {
+    super(args, privateData);
+
+    this._eventDefaults.click = this._handleClick.bind(this);
+  }
+
   get form() {
     return closest(this, "form");
   }
+
+  _handleClick(event) {
+    if (this.hasAttribute('for')) {
+      const node = this.ownerDocument.getElementById(this.getAttribute('for'));
+      if (node && this._isLabelable(node)) {
+        this._dispatchClickEvent(node);
+      }
+    } else if (this.children.length) {
+      const kids = Array.prototype.slice.call(this.querySelectorAll('*'));
+      while (kids.length) {
+        if (this._isLabelable(kids[0])) {
+          this._dispatchClickEvent(kids[0]);
+          break;
+        }
+
+        kids.shift();
+      }
+    }
+  }
+
+  _isLabelable(node) {
+    /* labelable logic defined at: https://html.spec.whatwg.org/multipage/forms.html#category-label */
+    if (node.nodeType !== 1) { return false; }
+
+    switch (node.tagName) {
+    case 'BUTTON':
+    case 'KEYGEN':
+    case 'METER':
+    case 'OUTPUT':
+    case 'PROGRESS':
+    case 'SELECT':
+    case 'TEXTAREA':
+      return true;
+
+    case 'INPUT':
+      return node.type !== 'hidden' ? true : false;
+    }
+
+    return false;
+  }
+
+  _dispatchClickEvent(node) {
```

Comment:
Also maybe name it "fire a click event" per https://html.spec.whatwg.org/#fire-a-click-event, and make sure that https://html.spec.whatwg.org/#fire-a-synthetic-mouse-event matches the details for the mouse event.